server/repositories/recommendations_repository.py

```diff
@@ -0,0 +1,115 @@
+import math
+import random
+from pypika import Query
+from sklearn.preprocessing import MinMaxScaler
+from sklearn.neighbors import NearestNeighbors
+
+from server.dbutils.map_rt import MapRT
+from server.repositories.base_repository import Repository
+from server.repositories.listing_repsotory import Listing as Listing_Repository
+
+
+class RecommendationsRepository(Repository):
+
+    def recommend_n_listings(self, listing_id, request_filter, n=5):
+        """ Recommend n Listings factoring in the provided filter criteria
+
+        :param listing_id: Listing ID of target
+        :param request_filter: filter criteria
+        :param n: number of recommendations to compute
+        :return: recommended Listings
+        :raises ValueError: if no ID or filter was provided or n < 1
+        """
+        # preconditions
+        if request_filter is None:
```

Comment:
Did some basic error handling, maybe there is a better way to do this in Python?